fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/client/event/lifecycle/v1/ClientLifecycleEvents.java

```diff
@@ -0,0 +1,196 @@
+				event.onTick(client);
+			}
+		}
+	});
+
+	public static final Event<WorldTickCallback<ClientWorld>> WORLD_TICK = EventFactory.createArrayBacked(WorldTickCallback.class, callbacks -> world -> {
+		if (EventFactory.isProfilingEnabled()) {
+			final Profiler profiler = world.getProfiler();
+			profiler.push("fabricClientWorldTick");
+
+			for (WorldTickCallback<ClientWorld> callback : callbacks) {
+				profiler.push(EventFactory.getHandlerName(callback));
+				callback.onTick(world);
+				profiler.pop();
+			}
+
+			profiler.pop();
+		} else {
+			for (WorldTickCallback<ClientWorld> callback : callbacks) {
+				callback.onTick(world);
+			}
+		}
+	});
+
+	public static final Event<ChunkLoadCallback<ClientWorld>> CHUNK_LOAD = EventFactory.createArrayBacked(ChunkLoadCallback.class, callbacks -> (clientWorld, chunk) -> {
+		if (EventFactory.isProfilingEnabled()) {
+			Profiler profiler = clientWorld.getProfiler();
+			profiler.push("fabricClientChunkLoad");
+
+			for (ChunkLoadCallback<ClientWorld> callback : callbacks) {
+				profiler.push(EventFactory.getHandlerName(callback));
+				callback.onChunkLoad(clientWorld, chunk);
+				profiler.pop();
+			}
+
+			profiler.pop();
+		} else {
+			for (ChunkLoadCallback<ClientWorld> callback : callbacks) {
+				callback.onChunkLoad(clientWorld, chunk);
+			}
+		}
+	});
+
+	public static final Event<ChunkUnloadCallback<ClientWorld>> CHUNK_UNLOAD = EventFactory.createArrayBacked(ChunkUnloadCallback.class, callbacks -> (clientWorld, chunk) -> {
+		if (EventFactory.isProfilingEnabled()) {
+			Profiler profiler = clientWorld.getProfiler();
+			profiler.push("fabricClientChunkUnload");
+
+			for (ChunkUnloadCallback<ClientWorld> callback : callbacks) {
+				profiler.push(EventFactory.getHandlerName(callback));
+				callback.onChunkUnload(clientWorld, chunk);
+				profiler.pop();
+			}
+
+			profiler.pop();
+		} else {
+			for (ChunkUnloadCallback<ClientWorld> callback : callbacks) {
+				callback.onChunkUnload(clientWorld, chunk);
+			}
+		}
+	});
+
+	public static final Event<BlockEntityLoadCallback<ClientWorld>> BLOCK_ENTITY_LOAD = EventFactory.createArrayBacked(BlockEntityLoadCallback.class, callbacks -> (blockEntity, world) -> {
+		final Profiler profiler = world.getProfiler();
+
+		if (EventFactory.isProfilingEnabled()) {
+			profiler.push("fabricClientBlockEntityLoad");
+
+			for (BlockEntityLoadCallback<ClientWorld> callback : callbacks) {
+				profiler.push(EventFactory.getHandlerName(callback));
+				callback.onLoadBlockEntity(blockEntity, world);
+				profiler.pop();
+			}
+
+			profiler.pop();
+		} else {
+			for (BlockEntityLoadCallback<ClientWorld> callback : callbacks) {
+				callback.onLoadBlockEntity(blockEntity, world);
+			}
+		}
+	});
+
+	public static final Event<BlockEntityUnloadCallback<ClientWorld>> BLOCK_ENTITY_UNLOAD = EventFactory.createArrayBacked(BlockEntityUnloadCallback.class, callbacks -> (blockEntity, world) -> {
+		final Profiler profiler = world.getProfiler();
```

Comment:
any point of getting the profiler before the profiler enabled check?

Reply:
Really nothing. Just wanted to throw the profiler into the variable stack and forget about it. I'll probably move it into the if block

Reply:
All the profiler calls should be within the `isProfilingEnabled` block now